apps/cowswap-frontend/src/modules/hooksStore/hooks/usePostHooksRecipientOverride.ts

```diff
@@ -0,0 +1,9 @@
+import { useMemo } from 'react'
+
+import { useHooks } from './useHooks'
+
+export function usePostHooksRecipientOverride() {
+  const { postHooks } = useHooks()
+
+  return useMemo(() => postHooks.reverse().find((i) => i.receiverOverride)?.receiverOverride, [postHooks])
```

Comment:
so we just consider the last hook recipient override 👍

Reply:
another minor thing I noticed: you've called the hook usePostHooks*Recipient*Override and the argument is *receiver*Override. I'm just now getting familiar with the cowswap codebase, but I'm curious to understand whether you prefer one term over the other since I've seen both used across the codebase.

Reply:
You are 100% right, there must be a validation an warning for the recipient override.
I haven't implemented it in this PR, but we would definetely should do it in the future.

Reply:
Added TODO: https://github.com/cowprotocol/cowswap/pull/4878/commits/adcc617cd93cb744f63b965dbe6de4d189ac21a1